### PR TITLE
Make read task snackbar toggleable

### LINKED ISSUE #230: Close Tasks by Re-clicking Task Icon
Rather than needing to click "close" in the snackbar to close a task, can we re-click the task icon in the top left to close the task snackbar? That way if someone clicks once to read they can quickly re-click again to close without moving their cursor across the screen... sometimes its the little things :)

### PR DESCRIPTION
**Pull request by DCH25**
## Details

- Read Task icon on the board now only queues a snackbar if there's none currently open 

Closes #230

### CODE REVIEW

**Review by l3n0ire — APPROVED** [review]

File: src/app/components/canvas/canvas.component.ts [hunk 1]
```diff
@@ -427,6 +427,11 @@ export class CanvasComponent implements OnInit, OnDestroy {
     const title = this.board.task.title ? this.board.task.title : 'No task created!';
     const message = this.board.task.message;
 
+    if (this.snackbarService.snackBar._openedSnackBarRef) {
```
Comment (Ammar-T) [inline src/app/components/canvas/canvas.component.ts:430]: Could we make a function inside snackbarService to see if there's a currently-opened snack bar instead? 

We should keep the 'snackBar' field private and inaccessible from other classes. I forgot to do this initially.

Comment (DCH25) [reply src/app/components/canvas/canvas.component.ts]: makes sense, just added this change

File: src/app/services/snackbar.service.ts [hunk 2]
```diff
@@ -63,6 +63,10 @@ export class SnackbarService implements OnDestroy {
     );
   }
 
+  snackbarIsOpen(): Boolean {
+    return Boolean(this.snackBar._openedSnackBarRef);
+  }
```
Comment (Ammar-T) [inline src/app/services/snackbar.service.ts:68]: Thanks for adding this! 

I don't think you need to convert it to 'Boolean' here - it's fine to just return a 'boolean'.

**Review by Ammar-T — APPROVED** [review]
Just one more small suggestion then feel free to merge!

**Pull request merged**